Lib/test/test_time.py

```diff
@@ -158,10 +158,45 @@ def test_conversions(self):
         self.assertEqual(int(time.mktime(time.localtime(self.t))),
                          int(self.t))
 
-    def test_sleep(self):
+    def test_sleep_exceptions(self):
+        self.assertRaises(TypeError, time.sleep, [])
+        self.assertRaises(TypeError, time.sleep, "a")
+        self.assertRaises(TypeError, time.sleep, complex(0, 0))
+
         self.assertRaises(ValueError, time.sleep, -2)
         self.assertRaises(ValueError, time.sleep, -1)
-        time.sleep(1.2)
+        self.assertRaises(ValueError, time.sleep, -0.1)
+
+    def test_sleep(self):
+        for value in [-0.0, 0, 0.0, 1e-6, 1, 1.2]:
+            with self.subTest(value=value):
+                time.sleep(value)
+
+    @unittest.skipIf(support.MS_WINDOWS, 'test only for non-Windows platforms')
+    def test_sleep_zero_posix(self):
```

Comment:
I dislike "performance" test. They tend to always fail on slowest buildbots.

Reply:
I can totally remove it. I think it was more important to have the type and value tests.